### PR TITLE
Rework our byte-buddy build integration to work with 1.10.15+

### PR DESCRIPTION
**Pull request by mcculls**
We can't use the default byte-buddy task because it assumes the
transformation's plugin class will be available from 'buildSrc'
before any modules have been built. Similarly our muzzle classes
make assumptions about what project types they can see from the
class loader that loaded them.

Instead we use our own shim that supports loading transformation
classes from modules in the build, like before. This also lets us
simplify the setup while still having proper task dependencies.

### CODE REVIEW

**Comment by tylerbenson** [comment]
Looks promising!

**Review by tylerbenson — APPROVED** [review]
It would be really great if you could figure out a unit test for the plugin.

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -23,6 +27,8 @@ dependencies {
   implementation(gradleApi())
   implementation(localGroovy())
 
+  implementation("net.bytebuddy", "byte-buddy-gradle-plugin", "1.11.1")
```
Comment (tylerbenson) [inline buildSrc/build.gradle.kts:30]: New version available: https://github.com/raphw/byte-buddy/releases/tag/byte-buddy-1.11.5

Comment (mcculls) [reply buildSrc/build.gradle.kts]: 👍   74a949361

**Review by bantonsson — APPROVED** [review]

**Comment by tylerbenson** [comment]
Thanks for adding the test @mcculls!

**Pull request merged**